frontend/src/main/java/com/google/tripmeout/frontend/storage/InMemoryTripModelStorage.java

```diff
@@ -0,0 +1,85 @@
+package com.google.tripmeout.frontend.storage;
+
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripAlreadyExistsException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class InMemoryTripModelStorage {
+  Map<String, TripModel> storage = new ConcurrentHashMap<>();
+
+  public void addTrip(TripModel trip) throws TripAlreadyExistsException {
+    synchronized(storage) {
```

Comment:
Nit: space after `synchronized` and before the opening `(`. The formatter _should_ cover this (I think).